crates/binjs_meta/src/util.rs

```diff
@@ -8,6 +8,14 @@ pub trait ToStr {
     fn newline(&self) -> String {
         format!("{}\n", self.to_str())
     }
+    /// Append newline if the string is not empty.
+    fn newline_if_not_empty(&self) -> String {
```

Comment:
Could you cache the result of `self.to_str()`? It can be rather expensive, in some cases.